### PR TITLE
Bump to 2024-3-1 and add docs for setup in editor, and new fetch in pyscript namespace.

### PR DESCRIPTION
**Pull request by ntoll**

### CODE REVIEW

File: docs/user-guide/builtins.md [hunk 1]
```diff
@@ -116,6 +116,74 @@ Such named modules will always then be available under the
...
+
+A common task is to `fetch` data from the web via HTTP requests. The
+`pyscript.fetch` function provides a uniform way to achieve this in both
+Pyodide and MicroPython. It is closely modelled on the
+[Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) found
+in browsers with some important Pythonic differences.
+
+The simple use case is to pass in a URL and `await` the response. Remember, in
+order to use `await` you must have the `async` attribute in the `script` tag
+that references your code.
```
Comment (WebReflection) [inline docs/user-guide/builtins.md:129]: or mark explicitly a generic `def` as `async` ?

File: docs/user-guide/builtins.md [hunk 2]
```diff
@@ -116,6 +116,74 @@ Such named modules will always then be available under the
...
+
+```python title="A simple HTTP GET with pyscript.fetch"
+from pyscript import fetch
+
+
+response = await fetch("https://example.com")
```
Comment (WebReflection) [inline docs/user-guide/builtins.md:135]: this is not going to work ... you need to explicitly state what do you want from it:

  * `await fetch(url).text()`
  * `await fetch(url).json()`
  * `await fetch(url).bytearray()`
  * `await fetch(url).arrayBuffer()`
  * `await fetch(url).blob()`

These are the options and how the fetch works.

Comment (ntoll) [reply docs/user-guide/builtins.md:135]: Ahhh... OK. I'll update.

File: docs/user-guide/builtins.md [hunk 3]
```diff
@@ -116,6 +116,74 @@ Such named modules will always then be available under the
...
+```
+
+If the `fetch` operation _returns a response that is not deemed `OK`_ (the
+definition of which
+[can be found here](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok))
+then an exception is raised.
+
+Assuming an `OK` response, the following methods are available to you to access
+the data returned from the server:
+
```
Comment (WebReflection) [inline docs/user-guide/builtins.md:145]: these are **all asynchronous** ... we removed the double await need but it's needed for any of these variants ... the `fetch` does pretty much nothing until you state what kind of result you are after ... once again:

  * await fetch(url).text()
  * await fetch(url).json()
  * await fetch(url).bytearray()
  * await fetch(url).arrayBuffer()
  * await fetch(url).blob()

That's it, don't split this in 2 operations or we offer nothing more than native `fetch` then.

Comment (WebReflection) [reply docs/user-guide/builtins.md:145]: P.S. eventually you can say you can:

```python
from pyscript import fetch

remote = fetch(url)

await remote.text()
# and all others
```

The target goal for the fetch is the most important and async thing we need and there's no workaround for that to be different, it's how every other *fetch* works out there, including the pyodide thing that requires double await ... I hope this is clear now, a bit surprised it wasn't already.

File: docs/user-guide/builtins.md [hunk 4]
```diff
@@ -116,6 +116,74 @@ Such named modules will always then be available under the
...
+* `response.blob()` returns a JavaScript [`blob`](https://developer.mozilla.org/en-US/docs/Web/API/Response/blob) version of the response. This is equivalent
+to the [`blob()` method](https://developer.mozilla.org/en-US/docs/Web/API/Response/blob) in the browser based `fetch` API.
+* `response.bytearray()` returns a Python [`bytearray`](https://docs.python.org/3/library/stdtypes.html#bytearray) version of the response.
+* `response.json()` returns a Python datastructure representing a JSON serialised payload in the response.
+* `response.text()` returns a Python string version of the response.
+
+!!! warning
+
+    Unlike the browser based `fetch` API, you **do not need to await** the
+    methods listed above, in order to get the data from the response.
```
Comment (WebReflection) [inline docs/user-guide/builtins.md:156]: wrong, that's the only thing you need to await here ... those are mandatory and async by all means, please see the counter example ... we offer one await but it's going to be a one-off operation.

Not sure why it's upside down here, please let's fix this.

Comment (ntoll) [reply docs/user-guide/builtins.md]: Done.

**Comment by ntoll** [comment]
Hey hey... thanks for the feedback. I've corrected my ill-founded assumptions.

**Review by WebReflection — APPROVED** [review]

**Pull request merged**